src/lib/dev-environment/dev-environment-cli.js

```diff
@@ -137,34 +133,48 @@ export async function promptForArguments( preselectedOptions: InstanceOptions, d
 		elasticsearch: preselectedOptions.elasticsearch || defaultOptions.elasticsearch || DEV_ENVIRONMENT_DEFAULTS.elasticsearchVersion,
 		mariadb: preselectedOptions.mariadb || defaultOptions.mariadb || DEV_ENVIRONMENT_DEFAULTS.mariadbVersion,
 		mediaRedirectDomain: preselectedOptions.mediaRedirectDomain || '',
-		wordpress: {},
-		muPlugins: {},
-		clientCode: {},
+		wordpress: {
+			mode: 'image',
+		},
+		muPlugins: {
+			mode: 'image',
+		},
+		clientCode: {
+			mode: 'image',
+		},
 		statsd: false,
 		phpmyadmin: false,
 		xdebug: false,
+		siteSlug: '',
 	};
 
 	if ( ! instanceData.mediaRedirectDomain && defaultOptions.mediaRedirectDomain ) {
 		const mediaRedirectPromptText = `Would you like to redirect to ${ defaultOptions.mediaRedirectDomain } for missing media files?`;
 		const setMediaRedirectDomain = await promptForBoolean( mediaRedirectPromptText, true );
 		if ( setMediaRedirectDomain ) {
-			instanceData.mediaRedirectDomain = defaultOptions.mediaRedirectDomain;
+			instanceData.mediaRedirectDomain = defaultOptions.mediaRedirectDomain ?? '';
 		}
 	}
 
 	for ( const component of DEV_ENVIRONMENT_COMPONENTS ) {
-		const option = preselectedOptions[ component ];
-		const defaultValue = defaultOptions[ component ];
+		const option = preselectedOptions[ component ].toString();
+		const defaultValue = ( defaultOptions[ component ] ?? '' ).toString();
```

Comment:
I wish we used TypeScript…

Value types of `InstanceOptions` properties can be strings and booleans. Flow does not like the `boolean` part, and it is hard to tell it that `boolean` won't happen here; `toString()` seems to be the least evil here.

The situation is much worse with `defaultValue`, as it is `$Shape<InstanceOptions>` and, therefore, we can have `undefined` here, which does not have a `toString()` method.

We have to use strings because this is what `processComponent()` expects.

Reply:
The error was

Cannot call `processComponent` with `option` bound to `preselectedValue` because  boolean [1] is incompatible with  string [2].

Cannot call `processComponent` with `defaultValue` bound to `defaultValue` because  boolean [1] is incompatible with  string [2].